CheoR/pack-it-up
# Cr refactors

# Description

Fixes bug where in Box detail page or Item detail page, when user goes to detail page, dropdown assignement did not show which move/box the box/item is currently assigned to.

When the user selected a new move/item, the drop down will revert to default option. 

Now dropdown retains selection state.

## Type of change


# Testing Instructions

`git fetch --all`
`git checkout cr-refactors`
`npm start`

In another tab, cd into `api` and run

`json-server -p 8088  database.json`

1. Create moves.
2. Create boxes and link them to moves.
3. Create items and link them to boxes.
4. Go either to the box detail or item detail page.
5. Dropdown select should show which move/box the box/time is assigned to.
6. Change assignment and udpate.
7. Dropdown should update to new selection.

# Checklist:
